File: java/src/main/java/no/bouvet/workshop/solid/CsvReader.java
package no.bouvet.workshop.solid;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;

public class CsvReader {

    public void parse(File file) throws IOException {
        System.out.println("Starting to parse: " + file);
        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
            String line;
            while ((line = reader.readLine()) != null) {
                String[] country = line.split(",");
                System.out.println("Country [code= " + country[4] + " , name=" + country[5] + "]");
            }
        }
        System.out.println("Done parsing: " + file);
    }
}

File: dotNet/delegation/delegation/CsvReader.cs
using System;
using System.IO;

namespace delegation
{
    public class CsvReader
    {
        public void parse(FileInfo file)
        {
            Console.WriteLine("Starting to parse: " + file);
            using (var reader = file.OpenText())
            {
                string line;
                while ((line = reader.ReadLine()) != null)
                {
                    string[] country = line.Split(',');
                    Console.WriteLine("Country [code= " + country[4] + " , name=" + country[5] + "]");
               }
            }
            Console.WriteLine("Done parsing: " + file);
        }
    }
}
